sources/api/ecs-settings-applier/build.rs

```diff
@@ -9,6 +9,17 @@ fn main() {
     println!("cargo:rerun-if-env-changed=VARIANT");
     if let Ok(variant) = env::var("VARIANT") {
         println!("cargo:rustc-cfg=variant=\"{}\"", variant);
+        let parts = variant.split('-').collect::<Vec<&str>>();
+        println!(
+            "cargo:rustc-cfg=variant_family=\"{}\"",
+            parts[0..2].join("-")
+        );
+        let variant_type = if parts.len() > 3 {
+            parts[3]
+        } else {
+            "general_purpose"
+        };
+        println!("cargo:rustc-cfg=variant_type=\"{}\"", variant_type);
```

Comment:
It doesn't have to be part of this PR, but we should generalize this logic. Possibly buildsys could have a lib.rs that we use for things like this.